### ISSUE TITLE
[a11y]: search/expandable missing tooltip

### ISSUE THREAD

**Issue by ariellalgilmore** [carbon-design-system/carbon#20772]
### Package

@carbon/react

### Browser

Chrome

### Package version

latest

### React version

latest

### Automated testing tool and ruleset

discussed here: https://github.com/carbon-design-system/carbon-labs/pull/854

### Description

Since search/expandable search are not using a Carbon button they do not automatically have a tooltip. Should we add one to the button here? 

In Carbon Labs we are using expandable search and it's not possible to add a tooltip currently

### Reproduction/example

https://deploy-preview-854--carbon-labs-react.netlify.app/?path=/story/components-uishell--demo

### Steps to reproduce

go to the link above and hover over the search there is no tooltip. Also, wrapping it in a HeaderGlobalAction causes issues because of the sizing restrictions

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
